framework/src/mysqlKVStore.js

```diff
@@ -0,0 +1,115 @@
+ *
+ */
+const { getTableInstance } = require('./mysql');
+const keyValueStoreSchema = require('./schema/kvStore');
+const Logger = require('./logger').get;
+
+const logger = Logger();
+
+const CONN_ENDPOINT_DEFAULT = 'mysql://lisk:password@localhost:3306/lisk';
+const KV_STORE_ALLOWED_VALUE_TYPES = ['boolean', 'number', 'bigint', 'string', 'undefined'];
+
+const formatValue = (value, type) => {
+	// Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#description
+	if (type === 'boolean') return Boolean(value);
+	if (type === 'number') return Number(value);
+	if (type === 'bigint') return BigInt(value);
+	if (type === 'string') return String(value);
+	if (type === 'undefined') return undefined;
+
+	// type: ['symbol', 'function', 'object'], should be unreachable
+	return value;
+};
+
+function createKeyValueTable(connEndpoint, prefix) {
+	const getKeyValueTableInstance = getTableInstance(keyValueStoreSchema, connEndpoint);
+
+	const set = async (key, value, dbTrx) => {
+		const keyValueTable = await getKeyValueTableInstance;
+		const type = typeof value;
+
+		if (!KV_STORE_ALLOWED_VALUE_TYPES.includes(type)) {
+			logger.error(`Allowed 'value' types are: ${KV_STORE_ALLOWED_VALUE_TYPES.join()}`);
+		}
+
+		const finalValue = value === undefined ? value : String(value);
+		await keyValueTable.upsert({ key: `${prefix}_${key}`, value: finalValue, type }, dbTrx);
+	};
+
+	const get = async (key, dbTrx) => {
+		const keyValueTable = await getKeyValueTableInstance;
+		const [{ value, type } = {}] = await keyValueTable.find(
+			{ key: `${prefix}_${key}`, limit: 1 },
+			['value', 'type'],
+			dbTrx,
+		);
+
+		return formatValue(value, type);
+	};
+
+	const getByPattern = async pattern => {
+		const keyValueTable = await getKeyValueTableInstance;
+		const result = await keyValueTable.find(
+			{ search: { property: 'key', pattern } },
+			['key', 'value', 'type'],
+		);
+
+		const formattedResult = result.map(row => ({
+			key: row.key.slice(prefix.length + 1),
+			value: formatValue(row.value, row.type),
+		}));
+		return formattedResult;
+	};
+
+	const deleteKey = async (key, dbTrx) => {
+		const keyValueTable = await getKeyValueTableInstance;
+		return keyValueTable.deleteByPrimaryKey([`${prefix}_${key}`], dbTrx);
+	};
+
+	return {
+		set,
+		get,
+		getByPattern,
+		delete: deleteKey,
+	};
+}
+
+function keyValueTableInstance(connEndpoint, prefix) {
+	if (connEndpoint && prefix) {
+		keyValueTableInstance.instance = createKeyValueTable(connEndpoint, prefix);
+	} else if (!keyValueTableInstance.instance) {
+		logger.error('Key value table not configured');
+		throw new Error('Key value table not configured');
+	}
+
+	return keyValueTableInstance.instance;
+}
+
+function configureKeyValueTable(connEndpoint = CONN_ENDPOINT_DEFAULT, prefix = 'default') {
+	return keyValueTableInstance(connEndpoint, prefix);
+}
+
+function getKeyValueTable() {
+	return keyValueTableInstance();
+}
```

Comment:
Lets use arrow functions for consistency throughout the codebase. And, let's discuss the implementation offline.